bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerHttpBridgeHandler.java

```diff
@@ -12,83 +12,131 @@
+            } catch (UnauthorizedApiException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        "Unauthorized, check your password is correct");
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    private void setupAPI() {
         logger.debug("Initializing OpenSprinkler with config (Hostname: {}, Port: {}, Refresh: {}).",
                 openSprinklerConfig.hostname, openSprinklerConfig.port, openSprinklerConfig.refresh);
-
-        OpenSprinklerApi openSprinklerDevice;
         try {
             openSprinklerDevice = apiFactory.getHttpApi(openSprinklerConfig);
-            this.openSprinklerDevice = openSprinklerDevice;
+            openSprinklerDevice.enterManualMode();
         } catch (CommunicationApiException | GeneralApiException exp) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    "Could not create API connection to the OpenSprinkler device. Error received: " + exp);
-
+                    "Could not create an API connection to the OpenSprinkler. Error received: " + exp);
             return;
         }
-
-        logger.debug("Successfully created API connection to the OpenSprinkler device.");
-
-        try {
-            openSprinklerDevice.enterManualMode();
-        } catch (CommunicationApiException exp) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
-                    "Could not open API connection to the OpenSprinkler device. Error received: " + exp);
-        }
-
-        if (openSprinklerDevice.isManualModeEnabled()) {
-            updateStatus(ThingStatus.ONLINE);
-        } else {
+        if (!openSprinklerDevice.isManualModeEnabled()) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                     "Could not initialize the connection to the OpenSprinkler.");
-
-            return;
         }
+    }
 
-        super.initialize();
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Nothing to do for the bridge handler
     }
 
     @Override
-    protected long getRefreshInterval() {
-        OpenSprinklerHttpInterfaceConfig openSprinklerConfig = this.openSprinklerConfig;
-        if (openSprinklerConfig == null) {
-            return DEFAULT_REFRESH_RATE;
+    public void initialize() {
+        openSprinklerConfig = getConfig().as(OpenSprinklerHttpInterfaceConfig.class);
+        pollingJob = scheduler.scheduleWithFixedDelay(this::refreshStations, 2, openSprinklerConfig.refresh,
+                TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        OpenSprinklerApi localApi = openSprinklerDevice;
+        if (localApi != null) {
+            try {
+                localApi.leaveManualMode();
```

Comment:
`dispose()` should return fast. Is this really necessary?

Reply:
I did not write the code and left it alone as I can not test with firmware that old. My personal opinion is that in any of the recent firmwares it is not necessary, and only in firmware that is years old would leaveManualMode() actually do anything. It could be removed and I doubt anyone would notice.

Reply:
I'd leave it if it breaks old firmwares.